docs: add instructions for how to connect to local database from docker container

sarthak opened the pull request:
### SUMMARY
Update installation instructions to include a section on
connecting to local db via superset running in docker container.

Newcomers may have trouble in this part because it involves
knowing about some docker internal details, as well as
google search doesn't bring many directly useful results.

If it's a commonly occurring issue among new comers,
we may accept this commit.

### TESTING INSTRUCTIONS
This is a documentation update, so no testing required except manual review.

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

sarthak committed:
Update installation instructions for docker

Update installation instructions to include a section on
connecting to local db via superset running in docker container.

Newcomers may have trouble in this part because it involves
knowing about some docker internal details, as well as
google search doesn't bring many directly useful results.

If it's a commonly occurring issue among new comers,
we may accept this commit.

thathert commented on docs/src/pages/docs/installation/index.mdx:
```diff
@@ -100,3 +100,11 @@ username: admin
+When running superset using `docker` or `docker-compose` it runs in its own docker container, as if the superset was running in a seperate machine entirely. Therefore attempts to connect to your local database with hostname `localhost` won't work as `localhost` refers to the docker container superset is running in, and not your actual host machine. Fortunately docker provides an easy way to acces ...
+
+1. Configuring the local postgresql/database instance to accept public incoming connections. By default postgresql only allows incoming connections from localhost only, but re-iterating once again, `localhosts` are different for host machine and docker container. For postgresql this involves make one-line changes to the files `postgresql.conf` and `pg_hba.conf`, you can find helpful links tailore ...
+2. Instead of `localhost`, try using `host.docker.internal` or `172.18.0.1` as the host name when attempting to connect to the database. This is docker internal detail, to find the exact host name you want to use you can also do `ifconfig` or `ip addr show` and look at the ip address of `docker0` interface that must have been created by docker for you.
```
You can also look at the gateway ip in the output of `docker network inspect bridge`

sarthak replied:
Added.

thathert reviewed (commented):
Thanks for adding this, I ran into the same thing when I was getting started and this would have saved a good amount of google-time.

sarthak commented on docs/src/pages/docs/installation/index.mdx:
```diff
@@ -100,3 +100,11 @@ username: admin
+
+When running superset using `docker` or `docker-compose` it runs in its own docker container, as if the superset was running in a seperate machine entirely. Therefore attempts to connect to your local database with hostname `localhost` won't work as `localhost` refers to the docker container superset is running in, and not your actual host machine. Fortunately docker provides an easy way to acces ...
+
+1. Configuring the local postgresql/database instance to accept public incoming connections. By default postgresql only allows incoming connections from localhost only, but re-iterating once again, `localhosts` are different for host machine and docker container. For postgresql this involves make one-line changes to the files `postgresql.conf` and `pg_hba.conf`, you can find helpful links tailore ...
```
It might be an OS thing. In linux first of all `host.docker.internal` does not work, and second it was necessary when used with `172.18.0.1` as hostname. Though, agreed that this is no more than a hack. Maybe we can add a *note* here for convenience but those managing production databases should already know better!

thathert replied:
Ah yes so maybe it'd be better to split this up. I think what it is, is that on mac you can do `host.docker.interal` and don't need to touch your postgresql confs. The behavior is described [here](https://docs.docker.com/docker-for-mac/networking/#there-is-no-docker0-bridge-on-macos).

On other OS you don't have the magic dns entry and so you need to go through the bridge network IP which likely requires the config updates.

sarthak committed:
Update index.mdx

Add a caution regarding changing database config to allow public incoming.
Add a note that mac users may skip configuring database.
Spelling mistake fix.
Style fix, "Superset" is now in title case.

sarthak commented on docs/src/pages/docs/installation/index.mdx:
```diff
@@ -100,3 +100,11 @@ username: admin
+
+### 5. Connecting your local database instance to superset
+
+When running superset using `docker` or `docker-compose` it runs in its own docker container, as if the superset was running in a seperate machine entirely. Therefore attempts to connect to your local database with hostname `localhost` won't work as `localhost` refers to the docker container superset is running in, and not your actual host machine. Fortunately docker provides an easy way to acces ...
```
Done.

thathert reviewed (approved):
Looks good to me, thanks for taking the time to add this!

srinify commented:
Hi @sarthak thanks for adding this! This has been on my list to add for a long time, glad you took the initiative.

srinify commented:
Hi @sarthak for future reference, your docs focused PRs need to be pre-fixed with `docs: ` for the PR linter to be happy!
